pkg/scheduler/objects/queue.go

```diff
@@ -1858,7 +1858,10 @@ func (sq *Queue) findPreemptionFenceRoot(priorityMap map[string]int64, currentPr
 	}
 	priorityMap[sq.QueuePath] = currentPriority
 
-	if sq.parent == nil || sq.GetPreemptionPolicy() == policies.FencePreemptionPolicy {
+	// Is FencePreemptionPolicy set on the queue?
+	// Does Queue's Usage already reached Max Res? If Yes, No use in picking up the victims outside this queue path, have to find the victim queue path with in this queue hierarchy.
+	// At last, root queue would be the fence if none met above
```

Comment:
This comment is a bit too verbose. i'd rephrase this eg. "Return this queue as fence root if: 1. it is marked as such 2. root queue 3. allocations in the queue reached maximum resources".